pxr/usd/plugin/usdAbc/alembicReader.cpp

```diff
@@ -3479,6 +3484,10 @@ _ReadPrim(
         // huge if statement or deep if nesting, we'll use do/while and
         // break to do it.
         do {
+            if (!TfGetEnvSetting(USD_ABC_XFORM_PRIM_COLLAPSE)) {
+                // Transform collapse is specified as unwanted behavior
+                break;
+            }
             // Parent has to be a transform.
```

Comment:
Hi Alan!  We're looking to accepting this PR shortly; one thing I'd like to point out, which is completely undocumented, is that TfGetEnvSetting() is protected by a std::mutex, so by adding a call to it in _ReadPrim(), we're potentially introducing contention if Alembic files are being opened simultaneously in multiple threads. (We should have made this same observation for the change that introduced USD_ABC_WRITE_UV_AS_ST_TEXCOORD2FARRAY, which we should fix).

What I'd suggest is adding a member flag to class _ReaderContext, which gets initialized in its ctor, then _ReadPrim() can just consult that flag.

Cheers,
--spiff

Reply:
Please ignore my prior comment, Alan - I was looking at the wrong version of TfGetEnvSetting - what you have should be fine - no changes necessary!

Sorry about that!
--spiff